src/test/java/net/sf/jabref/importer/fileformat/InspecImportTest.java

```diff
@@ -0,0 +1,109 @@
+package net.sf.jabref.importer.fileformat;
+
+import static org.junit.Assert.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.BibtexEntryTypes;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import org.junit.Before;
+import org.junit.Test;
+
+public class InspecImportTest {
+
+    private InspecImporter inspecImp;
+
+    @Before
+    public void setUp() throws Exception {
+        Globals.prefs = JabRefPreferences.getInstance();
+        this.inspecImp = new InspecImporter();
+    }
+
+    @Test
+    public void testIsRecognizedFormatAccept() throws IOException {
+        List<String> testList = Arrays.asList("InspecImportTest.txt", "InspecImportTest2.txt");
+        for (String str : testList) {
+            try (InputStream inStream = InspecImportTest.class.getResourceAsStream(str)) {
+                assertTrue(inspecImp.isRecognizedFormat(inStream));
+            }
+        }
+    }
+
+    @Test
+    public void testIsRecognizedFormatReject() throws IOException {
+        List<String> testList = Arrays.asList("CopacImporterTest1.txt", "CopacImporterTest2.txt",
+                "IEEEImport1.txt", "IsiImporterTest1.isi", "IsiImporterTestInspec.isi", "IsiImporterTestWOS.isi",
+                "IsiImporterTestMedline.isi", "RisImporterTest1.ris", "InspecImportTestFalse.txt");
+
+        for (String str : testList) {
+            try (InputStream inStream = InspecImportTest.class.getResourceAsStream(str)) {
+                assertFalse(inspecImp.isRecognizedFormat(inStream));
+            }
+        }
+    }
+
+    @Test
+    public void testCompleteBibtexEntryOnJournalPaperImport() throws IOException {
+        try (InputStream inStream = InspecImportTest.class.getResourceAsStream("InspecImportTest2.txt")) {
+            List<BibEntry> entries = inspecImp.importEntries(inStream, new OutputPrinterToNull());
+            assertEquals(1, entries.size());
+            BibEntry entry = entries.get(0);
+
+            assertEquals("The SIS project : software reuse with a natural language approach", entry.getField("title"));
+            assertEquals("Prechelt, Lutz", entry.getField("author"));
+            assertEquals("1992", entry.getField("year"));
+            assertEquals("Abstrakt", entry.getField("abstract"));
+            assertEquals("key", entry.getField("keywords"));
+            assertEquals("10000", entry.getField("journal"));
+            assertEquals("20", entry.getField("pages"));
+            assertEquals("19", entry.getField("volume"));
+            assertEquals(BibtexEntryTypes.ARTICLE.getName().toLowerCase(), entry.getType());
```

Comment:
In production code, I completely agree: To compare one BibEntry with another one, a BibEntry must provide an equals method. We should implement one in the near future. Same holds for a BibDatabase which basically is a List of BibEntries. 

In test code, such an equals method does not help as it can only either give you true or false. Because of this, one uses Assert.assertEquals("a", "b") instead of Assert.assertTrue("a".equals("b")). Hence, we need a more sophisticated comparison method which can also show the difference, giving reasons why it failed. I think, we also agree on this one. 

I think we have to distinguish between the representation of the expected output and the comparison. 

We can represent the expected output using the canonical format in a file, in a String or by using the API to build up a BibEntry. I think using a file is more maintainable, as even normal users could provide us with test cases - but I do not have a strong opinion about this. All three alternatives are OK. 

So it boils down to how to achieve the comparison of two entries, or even two databases (list of entries) as the importers import a file which can contain more than one entry. I can also live with a comparison method which takes in two lists of BibEntries and compares them internally (while I personally prefer the canonical serialization as well as one would see all differences between two BibEntries quickly instead of only getting the single difference which caused the test to fail without knowing what other fields were also the problem). But nevertheless, we should require and use such a comparison method to ensure that no additional (and suprising) fields are found during the import. I thought we had such a comparison method already available, but I can only find `BibtexEntryAssert.assertEquals(List<BibEntry> expected, List<BibEntry> actuals)` which uses the canonical format internally for the comparison instead of calling the BibEntry methods directly.